File: src/easy/MostCommonSafeWord.swift
import Foundation

func mostCommonSafeWord(_ paragraph: String, _ banned: [String]) -> String {
  var freq: [String: Int] = [:]
  let separators = "!?',;. "
  let words = paragraph.lowercased().split(whereSeparator: { separators.contains($0) }).map(
    String.init)

  for word in words {
    if !banned.contains(word) {
      freq[word, default: 0] += 1
    }
  }
  let sorted = freq.sorted { $0.value > $1.value }
  return sorted.first?.key ?? ""
}

File: src/main/java/easy/MostCommonSafeWord.java
package easy;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

public class MostCommonSafeWord {
  public String solve(String paragraph, String[] banned) {
    List<String> bannedStream = Arrays.asList(banned);
    HashMap<String, Integer> freq = new HashMap<>();
    String[] words = paragraph.toLowerCase().split("[!?'.,;\\s]+");
    for (String word : words) {
      if (!bannedStream.contains(word)) {
        freq.put(word, freq.getOrDefault(word, 0) + 1);
      }
    }

    return freq.entrySet().stream()
        .max((a, b) -> a.getValue() - b.getValue())
        .map(entry -> entry.getKey())
        .orElse("");
  }
}
